### ISSUE TITLE
Automator folder create under Windows not working

### ISSUE THREAD

**Issue by Zeromax** [contao/core-bundle#461]
this:
```php
str_replace(TL_ROOT . '/', '',
```
should be
this:
```php
str_replace(TL_ROOT . DIRECTORY_SEPARATOR, '',
```

![directory](https://cloud.githubusercontent.com/assets/1867177/13860110/06045b26-ec87-11e5-9358-bf4e244d6907.PNG)

**Comment by Zeromax** [comment]
Files:
* vendor\contao\core-bundle\src\Resources\contao\library\Contao\Automator.php
* vendor\contao\core-bundle\src\Resources\contao\classes\PurgeData.php

I think all `str_replace(TL_ROOT . '/', '',` should be refactored to `str_replace(TL_ROOT . DIRECTORY_SEPARATOR, '',`

**Comment by Zeromax** [comment]
Or to ensure that all path have `/` instead of `\`

**Comment by leofeyer** [comment]
No, that would be wrong, because most paths will be normalized automatically by Symfony or the `FilesystemIterator::UNIX_PATHS` flag. We have had this problem already and it was you who reported it :smile: 

contao/core#7847

**Comment by Zeromax** [comment]
Aye... You are right. So I do not understand why this is broken in Contao 4?

But your Commit does not touch the automator... I am a bit confused

**Comment by fritzmg** [comment]
> No, that would be wrong, because most paths will be normalized automatically by Symfony

But they are not. Here are some `var_dump`s within the `Automator::purgePageCache` function:
```php
var_dump(\System::getContainer()->getParameter('kernel.cache_dir'));
var_dump(TL_ROOT . '/');
var_dump($strCacheDir);
```
»
```
string(37) "C:\xampp\htdocs\c41dev\app\cache\prod"
string(23) "C:\xampp\htdocs\c41dev/"
string(37) "C:\xampp\htdocs\c41dev\app\cache\prod"
```
Since you are using the wrong directory separator, the following code does not remove the root directory under Windows:
```php
$strCacheDir = str_replace(TL_ROOT . '/', '', \System::getContainer()->getParameter('kernel.cache_dir'));
```

**Comment by fritzmg** [comment]
> No, that would be wrong, because most paths will be normalized automatically by Symfony or the FilesystemIterator::UNIX_PATHS flag.

According to symfony/symfony#15472 Symfony should, in general, either return paths with only forward or only backward slashes, depending on the OS. If some functions are inconsistent to that rule, they need to be reported.

I didn't create another issue as requested there since I wasn't sure if the problem with `getRootDir()` actually still exists.

**Comment by leofeyer** [comment]
This seems to be a duplicate of #464 and should be fixed in 5040b0fde75bdeddafedc486233294c411ab73cf, too.
